src/screens/MapScreen/MapController.ts

```diff
@@ -1,55 +1,66 @@
-import type { ScreenSwitcher } from "../../types";
+import type { ScreenSwitcher, MapScreenConfig } from "../../types";
 import { ScreenController } from "../../types";
 import { MapScreenView } from "./MapView";
+import { map1Config, map2Config } from "../../configs/maps/MapScreenConfig.ts";
 
 export class MapScreenController extends ScreenController {
   private view: MapScreenView;
   private screenSwitcher: ScreenSwitcher;
+  private currentConfig: MapScreenConfig;
 
-  constructor(screenSwitcher: ScreenSwitcher) {
+  constructor(screenSwitcher: ScreenSwitcher, mapId: number = 1) {
     super();
-    this.view = new MapScreenView(
-      () => this.handleReferenceClick(),
-      () => this.handleRulesClick(),
-      () => this.handleExitClick(),
-      (level: string) => this.handleNodeClick(level),
-    );
     this.screenSwitcher = screenSwitcher;
+    this.currentConfig = this.getConfigForMap(mapId);
+    this.view = MapScreenView.fromConfig(
+      this.currentConfig,
+      (buttonId: string) => this.handleButtonClick(buttonId),
+      (nodeId: string) => this.handleNodeClick(nodeId),
+    );
   }
 
-  private handleNodeClick = (level: string) => {
-    switch (level) {
-      case "1":
-        this.screenSwitcher.switchToScreen({
-          type: "topic",
-          level: "friction",
-        });
-        break;
-      case "2":
-        this.screenSwitcher.switchToScreen({
-          type: "topic",
-          level: "projectile motion",
-        });
-        break;
-      case "Game 1":
-        this.screenSwitcher.switchToScreen({
-          type: "minigame",
-          screen: "title",
-          level: 1,
-        });
+  private getConfigForMap(mapId: number): MapScreenConfig {
+    switch (mapId) {
+      case 2:
+        return map2Config;
+      case 1:
+      default:
+        return map1Config;
```

Comment:
```suggestion
    switch (mapId) {
      case 1:
        return map1Config;
      case 2:
        return map2Config;
      default:
        return;
```

Just a little concern about the logic. The default behavior is to get configs for map 1. However, the default behavior can just to do nothing. What do you think?